src/analyses/base.ml

```diff
@@ -2026,6 +2026,80 @@ struct
+      | a when not (AD.is_top a) ->
+        AD.for_all (function
+            | Addr (v, _) -> ctx.ask (Queries.IsHeapVar v)
+            | _ -> false
+          ) a
+      | _ -> false
+    in
+    if points_to_heap_only then
+      (* Ask for BlobSize from the base address (the second field set to true) in order to avoid BlobSize giving us bot *)
+      ctx.ask (Queries.BlobSize {exp = dest; base_address = true})
+    else
+      match ctx.ask (Queries.MayPointTo dest) with
+      | a when not (Queries.AD.is_top a) ->
+        let pts_list = Queries.AD.elements a in
+        let pts_elems_to_sizes (addr: Queries.AD.elt) =
+          begin match addr with
+            | Addr (v, _) ->
+              begin match v.vtype with
+                | TArray (item_typ, _, _) ->
+                  let item_typ_size_in_bytes = (bitsSizeOf item_typ) / 8 in
+                  let item_typ_size_in_bytes = intdom_of_int item_typ_size_in_bytes in
+                  begin match ctx.ask (Queries.EvalLength dest) with
+                    | `Lifted arr_len -> `Lifted (ID.mul item_typ_size_in_bytes arr_len)
+                    | `Bot -> `Bot
+                    | `Top -> `Top
+                  end
+                | _ ->
+                  let type_size_in_bytes = (bitsSizeOf v.vtype) / 8 in
+                  `Lifted (intdom_of_int type_size_in_bytes)
+              end
+            | _ -> `Top
+          end
+        in
+        (* Map each points-to-set element to its size *)
+        let pts_sizes = List.map pts_elems_to_sizes pts_list in
+        (* Take the smallest of all sizes that ptr's contents may have *)
+        begin match pts_sizes with
+          | [] -> `Bot
+          | [x] -> x
+          | x::xs -> List.fold_left (fun acc elem ->
+              if ValueDomainQueries.ID.compare acc elem >= 0 then elem else acc
+            ) x xs
+        end
+      | _ ->
+        M.warn "Pointer %a has a points-to-set of top. An invalid memory access might occur" d_exp dest;
+        `Top
+
+  (* Used for memset() and memcpy() out-of-bounds checks *)
+  let check_count ctx fun_name dest n =
+    let (behavior:MessageCategory.behavior) = Undefined MemoryOutOfBoundsAccess in
+    let cwe_number = 823 in
+    let dest_size = get_min_size_of_dest ctx dest in
+    let eval_n = ctx.ask (Queries.EvalInt n) in
+    match ValueDomainQueries.ID.is_top dest_size, ValueDomainQueries.ID.is_top eval_n with
+    | true, _ ->
+      AnalysisState.svcomp_may_invalid_deref := true;
+      M.warn ~category:(Behavior behavior) ~tags:[CWE cwe_number] "Size of dest %a in function %s is unknown. Memory out-of-bounds access might occur" d_exp dest fun_name
+    | _, true ->
+      AnalysisState.svcomp_may_invalid_deref := true;
+      M.warn ~category:(Behavior behavior) ~tags:[CWE cwe_number] "Count parameter, passed to function %s is unknown. Memory out-of-bounds access might occur" fun_name
+    | false, false ->
+      if dest_size < eval_n then begin
```

Comment:
Is this a polymorphic comparison of things of type `ID.t` here? Then it's not what we want: We want to check if it may be smaller here, which may e.g. be done by `ID.lt`.

Reply:
Thanks for the tip! Should be fixed in https://github.com/goblint/analyzer/pull/1197/commits/fae72563a48b8b7a344395e8f6d75d76630bc8c9